includes/admin/class-llms-admin-profile.php

```diff
@@ -0,0 +1,269 @@
+<?php
+/**
+ * Handle extra profile fields for users in admin
+ *
+ * @package LifterLMS/Admin/Classes
+ *
+ * @since [verson]
+ * @version [version]
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Handle extra profile fields for users in admin
+ *
+ * Applies to edit-user.php, & profile.php.
+ *
+ * @since [version]
+ */
+class LLMS_Admin_Profile {
+
+	/**
+	 * Array of user profile fields
+	 *
+	 * @var array
+	 */
+	private $fields;
+
+	/**
+	 * Submission error
+	 *
+	 * @var null|WP_Error
+	 */
+	private $errors;
+
+	/**
+	 * Constructor
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function __construct() {
+
+		add_action( 'show_user_profile', array( $this, 'add_customer_meta_fields' ) );
+		add_action( 'edit_user_profile', array( $this, 'add_customer_meta_fields' ) );
+
+		add_action( 'personal_options_update', array( $this, 'save_customer_meta_fields' ) );
+		add_action( 'edit_user_profile_update', array( $this, 'save_customer_meta_fields' ) );
+
+		// Allow errors to be output.
+		add_action( 'user_profile_update_errors', array( $this, 'add_errors' ), 10, 3 );
+
+	}
+
+	/**
+	 * Add customer meta fields to the profile screens
+	 *
+	 * @since [version]
+	 *
+	 * @param WP_User $user Instance of WP_User for the user being updated.
+	 * @return void
+	 */
+	public function add_customer_meta_fields( $user ) {
+
+		if ( ! $this->current_user_can_manage_user_custom_fields( $user ) ) {
+			return;
+		}
+
+		/**
+		 * Enqueue select2 scripts and styles.
+		 */
+		wp_enqueue_script( 'llms-metaboxes' );
+		wp_enqueue_script( 'llms-select2' );
+		llms()->assets->enqueue_style( 'llms-select2-styles' );
+
+		$fields = $this->get_fields();
+
+		include_once LLMS_PLUGIN_DIR . 'includes/admin/views/user-edit-fields.php';
+
+	}
+
+	/**
+	 * Undocumented function
+	 *
+	 * Only who can 'manage_lifterlms' will be able to edit own and other profiles.
+	 *
+	 * @param WP_User $user Instance of WP_User for the user being updated.
+	 * @return boolean
+	 */
+	private function current_user_can_manage_user_custom_fields( $user ) {
+
+		return current_user_can( 'manage_lifterlms' ) && current_user_can( 'edit_users', $user );
+
+	}
+
+	/**
+	 * Maybe print validation errors
+	 *
+	 * @since [version]
+	 *
+	 * @param obj     $errors Instance of WP_Error, passed by reference.
+	 * @param bool    $update `true` if updating a profile, `false` if a new user.
+	 * @param WP_User $user   Instance of WP_User for the user being updated.
+	 * @return void
+	 */
+	public function add_errors( &$errors, $update, $user ) {
+
+		if ( is_wp_error( $this->errors ) && $this->errors->has_errors() ) {
+			foreach ( $this->errors->get_error_messages() as $message ) {
+				$errors->add(
+					'',
```

Comment:
@eri-trabiccolo that sounds good :+1:

Reply:
And you were right on this too.
I didn't feel to duplicate `copy_errors()` as  backward compat code but use the actual `copy_errors()` with wp 5.6+ since we still have to "parse" our error messages to make them look more like the wp ones:
`'<strong>Error</strong>: ...'` e.g. https://github.com/WordPress/WordPress/blob/de519f9dfefa1a10c93510c86b6cb941ae168ce8/wp-includes/user.php#L136-L141